types/client-data.interface.ts

```diff
@@ -7,11 +7,17 @@ interface GetResponse {
     version: any;
 }
 
+interface DeleteResponse {
+    success: boolean;
+    value: any;
```

Comment:
I wonder if we be a bit more specific here, i.e. using a `Record<string, any>` ? @yuhgto WDYT?

Reply:
I would also like having sth like `Record` here. I also proposed this for another place in the sdk (see my [community comment](https://community.monday.com/t/typescript-support-in-the-sdk/52646/6?u=j-ittermann-decadis)). Maybe this could be somehow generalized with some global type so that it could also be reused in the other places, e.g.
```ts
type ResponseData = Record<string, any>;
// ...
interface DeleteResponse {
    success: boolean;
    value: ResponseData;
}
// ...
api(query: string, options?: APIOptions): Promise<{ data: ResponseData}>;
```

But for sure questionable, if this is the scope of this PR here..

Reply:
Since the PR is all about typings, IMO this is the scope for this PR. No need to generalize the Record for other responses here though. Before we go about that, I'd like to get also @yuhgto's opinion as he is well versed in what data types are returned in various cases for this method

Reply:
Sorry to be the bearer of bad news, but I did some testing and found all sorts of problems here...

The docs don't fully describe the return values of these functions. So using the docs to create the types is problematic. To be fair, @zivnadel there's no way you would have known that! 

**What the response really looks like:** 
Here is a sample response when you call the deleteItem method: 
```json
{
  "method" : "storage",
  "data" : {
    "value" : null,
    "success" : true
  },
  "requestId" : "npzsegl"
}
```
The `setResponse` and `getResponse` types are also wrong, but it's out of the scope of this PR. 

**Possible fixes**
Depends on what we want to achieve here. And heads up: I know the SDK well but I don't have a deep knowledge of TS. 

**If we want to only describe the shape of the data:** use an interface
```typescript
interface DeleteResponse {
    method: string, 
    data: Record<string, any>,
    requestId: string
}
```

**If we want to have strong type checking with minimal headache if things change**: keep it simple with a record
``` typescript
type DeleteResponse = Record<string, any> 
```

**If we want both:** perhaps something like this. I'm new to TS so not sure if this is the best way to do it, but I know it works 😆 
```typescript
type DeleteKey = 'method' | 'data' | 'requestId';

type DeleteResponse = Record<string, any> 
```

Side note: @eidermaen great comment in the community about the return type of the API method! I see what you mean and agree :)

Reply:
Hi @yuhgto, thanks for your review! 😄  Please note what the `DeleteResponse` as well as `GetResponse` actually are. It is representing the **data** attribute of the response, and not the entire response:
```
deleteItem(key: string): Promise<{ data: DeleteResponse }>
```
I don't know if thats intentional, but I just copied the behavior of the existing types. So the interfaces aren't wrong, but misleading. I think they should be merged with bigger interface in that form:
```
interface DeleteResponse {
    method: string;
    data: {
        value: any; (or Record<string, any>)
        success: boolean;
    };
    requestId: string;
}
```

For some reason, `setItem` is acting differently (the `SetResponse` interface is used for the entire response and not only the data attribute):
```
setItem(key: string, value: any): Promise<SetResponse>;
```
I don't know if thats intentional, or the types are wrong.

About the docs. I didn't think I can't rely on them. I think they should be updated.

Also, review #83. It is related to this file and types.

Reply:
Oops my bad! Didn't realize you were defining only data part. 

Agreed, let's make a big interface that describes the shape of the whole return value. Spoke to @gregra81 and he agrees. Can you implement it this way?

Should be something like this:
```typescript
deleteItem(key: string): Promise<DeleteResponse>;

interface DeleteResponse {
    method: string;
    data: {
        value: any; 
        success: boolean;
    };
    requestId: string;
}
```

Anyway because GetItem and SetItem types are inconsistent, we can pick which pattern we like more :) 

I’ll make a PR later to add the missing keys to SetItem and GetItem.

Reply:
Done! Let me know if the changes look good and I'll approve + merge :)

Reply:
@yuhgto Look good, thanks!